curvefs/src/client/s3/client_s3_cache_manager.cpp

```diff
@@ -506,14 +503,225 @@ void FileCacheManager::PrefetchS3Objs(std::vector<std::string> prefetchObjs) {
             context->len = blockSize;
             context->cb = AsyncPrefetchCallback{inode, s3ClientAdaptor};
             VLOG(9) << "prefetch start: " << context->key
-                       << ", len: " << context->len;
+                    << ", len: " << context->len;
             s3ClientAdaptor->GetS3Client()->DownloadAsync(context);
         };
         s3ClientAdaptor_->PushAsyncTask(task);
     }
     return;
 }
 
+void FileCacheManager::GenerateS3Request(ReadRequest request,
+                                         const S3ChunkInfoList &s3ChunkInfoList,
+                                         char *dataBuf,
+                                         std::vector<S3ReadRequest> *requests,
+                                         uint64_t fsId, uint64_t inodeId) {
+    uint64_t blockSize = s3ClientAdaptor_->GetBlockSize();
+    uint64_t chunkSize = s3ClientAdaptor_->GetChunkSize();
+    std::map<uint64_t, ReadRequest> readRequests;  // first is chunkPos
+
+    VLOG(9) << "GenerateS3Request start request index:" << request.index
+            << ",chunkPos:" << request.chunkPos << ",len:" << request.len
+            << ",bufOffset:" << request.bufOffset;
+    readRequests.emplace(request.chunkPos, request);
+    int i = s3ChunkInfoList.s3chunks_size() - 1;
+    for (; i >= 0; i--) {
+        S3ChunkInfo s3ChunkInfo = s3ChunkInfoList.s3chunks(i);
+        uint64_t chunkInfoOffset = s3ChunkInfo.offset();
```

Comment:
No need to assign, just use s3ChunkInfo.offset() and s3ChunkInfo.len()

Reply:
avoid too many function calls